### DISCUSSION TITLE
request.all() additional data attributes

### DISCUSSION THREAD

**Discussion by WeyIin** [adonisjs/core#1687] [Help]
So I have a text box:
`<input type="text" id="item" name="item" data-item="5" data-len="3" class="hidden" />`

`request.all()` returns  `item: [ '' ]` (because it does not have set value)

Is there any way of retrieving additional data item from an input (data-*) ?

**Comment by nickngqs** [comment]
I dont think you're supposed to receive values for data- attributes. I recommend you to use axios to post your request to the server side.

You're supposed to have a payload with your http request

Frontend - Client side
```
const $item = document.getElementById('item')
const id = $item.dataset.item // or $item.getAttribute('data-item')
const len = $item.dataset.len // or $item.getAttribute('data-len')

axios.post('/route-name', { id, len });
```

Server - Adonis side
```
const { id, len } = response.post() // or response.all()
// log id = 5, len = 3
```
